### DISCUSSION TITLE
Internet Explorer 11 and Polyfills for Bootstrap 5

### DISCUSSION THREAD

**Discussion by ray73864** [twbs/bootstrap#31222] [General]
Originally, a long time ago, BS5 was going to support IE11, I see this has now been dropped :(

I have no choice but to continue support IE11 in the corporate environment... and because Chrome / Firefox, etc... don't work nicely in a terminal services environment where the same user could be logged in multiple times.

Was hoping to use BS5 because it was still at least going to support IE11, fair enough, all our stores use the new Edge or Chrome (for the most part, but they are easier to handle, it's mostly our admin people), and because I wanted to be gone from jQuery, i'm currently using BS4 in a jquery-less way, though it did mean sacrificing certain things (such as the awesome Bootbox extension).

Will I still be able to use BS5 with IE11 by including certain polyfills? And if so, which ones, as the documentation for BS5 doesn't mention any, it just flat out says no to IE11.

Would really like to be able to move forward (Again, don't bother saying 'just use Chrome, etc...' because I have zero choice in the matter, ZERO choice :( ).

More than happy to put in place polyfills, just need to know what ones I need.

The only javascript related thing I actually need of bootstrap is the Modal stuff so that I can make BootboxJS work, because honestly, BootboxJS is the best alert/confirm/prompt/dialog library I have found, have tried ones like Vanilla-Modal and they just all plain suck.

**Comment by ffoodd** [comment]
Thinking roughly, the main concern about making BS5 support IE11 is that we now use CSS custom properties (and will rely on them more and more). So you'd probably need to re-compile our CSS using [PostCSS Custom Properties](https://github.com/postcss/postcss-custom-properties)—for examples.

And obviously, adding IE11 to your BrowserList, so that AutoPrefixer take care of what it can.

There might be more to do, but poylfilling the JS part is pretty easy (with services like polyfill.io for example)—but CSS is the harder part.

You may find out more deeply what changed with dropping IE11 by reading the dedicated PR: #30377

**Comment by coliff** [comment]
I'm maintaining an open-source project for making Bootstrap 5 compatible with IE11.
https://github.com/coliff/bootstrap-ie11

It fixes:
- Polyfill for CSS variables courtesy of [ie11CustomProperties](https://github.com/nuxodin/ie11CustomProperties)
- Workaround for the SVG overflow bug
- Remove the default vertical scrollbar from textarea
- Fixes for card image size bug
- Fixes for modals (.modal-dialog-centered and .modal-dialog-scrollable)
- Fixes for navbars (using flex for .navbar-collapse)
- Adds -ms- vendor prefixes for grid layout utilities (.align-self-auto, .align-self-center and .align-self-stretch)

It's already working really well - all the JavaScript components are 100% working and the CSS and all the docs and examples are looking good. The only remaining issue I have is for the custom checkboxes and switches which I expect to finish next week.
You can test it in IE11 at: https://coliff.github.io/bootstrap-ie11/tests/

**Reply by ray73864** [reply]
Awesome news, thanks for that :D

**Reply by amigoow** [reply]
Thanks so much @coliff! Appreciate that.. FYI, for some reason your demo domain is down: https://coliff.github.io/bootstrap-ie11/tests/

Just thought to let you know.

**Reply by coliff** [reply]
heya @amigoow - thanks for the message. The demo domain is working for me - can you pls try again and if issue persists please open an issue on the repo. thanks!

![image](https://user-images.githubusercontent.com/1212885/179164505-676440db-3a5e-4127-b255-700db5ddd995.png)
